src/entries/popup/components/CommandK/utils.ts

```diff
@@ -427,3 +430,61 @@ export function useKeyboardNavigation(
     };
   }, []);
 }
+
+interface WalletObj {
+  accounts: Address[];
+  type: 'HdKeychain' | 'HardwareWalletKeychain' | 'ReadOnlyKeychain';
+  vendor?: string;
+}
+
+type WalletNames = {
+  [address: string]: string;
+};
+
+const typeMapping: { [key: string]: string } = {
+  ReadOnlyKeychain: 'Watching',
+  HdKeychain: 'Imported',
+};
+
+export const generateCSV = (
+  data: WalletObj[],
+  walletNames: WalletNames,
+): string => {
+  let csvContent = 'public_address,name,type\n';
+
+  data.forEach(({ accounts, type, vendor }) => {
+    const address = accounts[0];
+    const name = walletNames[address as Address] || address;
+
+    let importedType = typeMapping[type];
+
+    if (type === 'HardwareWalletKeychain') {
+      importedType = `Hardware Wallet - ${vendor}`;
```

Comment:
To deobfuscate a user with multiple hardware devices, what if keep this as `Hardware Wallet` and then add an additional column called `device` that would have the manufacturer and the device ID (i.e. #1) so `Ledger #1` and `Ledger #2` and `Trezor #1`. IIRC those number ids / UUID should be available in the keychain

Reply:
Otherwise working well and looking good. Nice icon picks 👌

Reply:
This i think requires a few bigger changes to keychain things so will look at this in a followup PR to keep scope here reasonable.